### DISCUSSION TITLE
UISelectionList event UI_SELECTION_LIST_NEW_SELECTION return object_id of the UIWindow, not the selection item id that was added

### DISCUSSION THREAD

**Discussion by wadsworj** [MyreMylar/pygame_gui#276] [Q&A]
I've created a (UIWindow) container:

```
        super().__init__(pygame.Rect(self.position, self.size), self.ui_manager,
                         window_display_title=title,
                         object_id='#pong_window',
                         resizable=True)
```
then I created a UISelectionList:
```
        self.test_drop_down_menu = UISelectionList(pygame.Rect(10, 10, selection_list_size[0], selection_list_size[1]),
                                                   item_list=elements,
                                                   manager=self.ui_manager,
                                                   container=self)
```
Next I build a list of tuples with the format (str(text), str(object_id)) and assign this list of tuples to the UISelectionList:
`self.test_drop_down_menu.set_item_list(elements)`

When I get the event of a selection list item being selected, as such:

```
            if event.type == pygame_gui.UI_SELECTION_LIST_NEW_SELECTION and event.ui_element == self.test_drop_down_menu:
                self.presenter.handle_element_click(event.ui_object_id)
```
the event.ui_object_id is not the id I had assigned in the tuple when calling set_item_list

![image](https://user-images.githubusercontent.com/5513838/174327287-7ae20ef3-c352-4a77-975b-2b1eba981732.png)

how do I identify which of the selection items is firing the event? Is this the correct ID I am getting back? I would like to access 'str(object_id)'

After writing all of this out, I suppose my next step would be to get the selection item based on the text returned, which I should then be able to get the original id. But this seems clunky.

**Comment by MyreMylar** [comment]
Hello,

Looking over the code history, I know why this exists as it does. Originally the `UIButton` items in a selection list couldn't have unique object IDs - they all just got a default ID - which is still the behaviour now if you just pass in a list of strings when creating the `UISelectionList`. When the functionality was added to pass in strings and IDs it was just a small part of a larger refactor to use UISelectionLists in the UIDropDown. 

You can see the code I added to the drop down to do what you are trying to do here:

```
if (event.type == UI_SELECTION_LIST_NEW_SELECTION and
                event.ui_element == self.options_selection_list):
            selection = self.options_selection_list.get_single_selection()
```

Which is technically correct to spec as it is the `UISelectionList` firing off this event, not the `UIButton` itself, but I can see how in other use cases (definitely multiple selection lists) it would be useful to have better details on the actual button and the specific ID of the button that causes this event to be fired by the list.

I think it might be most useful to add two new bits of info to the event dictionary for the three `UI_SELECTION_LIST_` events:

1. the actual button itself as `.selected_item_element`.
2. the button's `.most_specific_combined_id` as `.selected_item_object_id` which would then have the theming ID you supply at the start when creating the `UISelectionList` as the last part of a string joined by full stops.

That way we should be fully backwards compatible. Does that sound sensible to you?
